src/util/dynamic_config.rs

```diff
@@ -107,3 +107,57 @@ impl<
         self.data.load().clone()
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::io::Write;
+
+    use super::*;
+    use tempfile::{NamedTempFile};
+    use serde::{Deserialize};
+
+    #[derive(Serialize, Deserialize, Clone)]
+    struct ConfigYaml {
+        pub foo: String
```

Comment:
is struct ConfigYaml(String) better?

Reply:
Could you explain what "struct ConfigYaml(String)" is? I don't know we can add a constructor for struct. We only need to desrialize ConfigYaml from file. 

I do wonder if there's better way to expect panic from test_invalid_yaml. Any suggestion?

Reply:
struct ConfigYaml(pub String);

this is a tuple struct
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=a52a6ce8c977b2eaae0c22c0489fb2bb

Reply:
see if #[should_panic] is the one you want
https://doc.rust-lang.org/book/ch11-01-writing-tests.html

Reply:
> tuple struct

Thanks! it doesn't seem to be able to deserialize from YAMl to tuple struct properly since the field doesn't have name.

Reply:
> see if #[should_panic] is the one you want https://doc.rust-lang.org/book/ch11-01-writing-tests.html

Yes, somehow my first try did work :-) 

Done and PTAL